app/models/user.rb

```diff
@@ -7,7 +7,47 @@ class User < ActiveRecord::Base
          :timeoutable, :confirmable
   include DeviseTokenAuth::Concerns::User
 
+  attr_accessor :initiate_email_reconfirmation
+
   def error_details(field)
     errors.details[field].collect { |i| i[:error] }
   end
+
+  def email_changed?
+    initiate_email_reconfirmation.present?
+  end
+
+  # override from Devise so that we can add hook into Salesforce update
+  def self.confirm_by_token(confirmation_token)
```

Comment:
first part of your question -- yes. 

second part, for password changing we don't need to do anything with salesforce though, right?